app/parsers/bulkrax/csv_parser.rb

```diff
@@ -209,11 +210,30 @@ def key_allowed(key)
 
     # All possible column names
     def export_headers
-      headers = ['id']
-      headers << source_identifier.to_s
-      headers << 'model'
-      headers << 'collections'
-      importerexporter.mapping.each_key { |key| headers << key if key_allowed(key) }
+      headers = []
+      # the object keys are not part of a work types default metadata but the object itself is,
+      # so we check according to it and add the acceptable headers
+      importerexporter.mapping.each do |key, value|
+        if value.key?('object') && key_allowed(value['object'])
+          object_keys = self.headers.each { |sh| sh.to_s.include?(value['object']) }
+          next object_keys.each { |item| headers << item }
```

Comment:
why do you need `next` here? doesn't it do object.keys.each {|item| headers << item} all the time?

Reply:
my intention was so that the code on line 222 didn't get run. `key_allowed(key)` would return false anyway for an object field mapping so the header wouldn't be added, but I figured no point having the code run.

unless the `next` has no bearing on line 222 since it's out of the loop?